transcom/mymove
# B 19894 too queue moves int

> [!IMPORTANT]  
> The large number of commits here are a result of bringing in the latest main.
> See main branch for commits to pay attention to as part of 19894 work: https://github.com/transcom/mymove/compare/main...B-19894-TOO-queue-moves-MAIN

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19894)

## Summary

This work is to show only moves with the following statuses in the TOO queue:
- `New Move`
- `Service Counseling Completed`
- `Approvals Requested` 

Previously, the TOO queue had filter status dropdowns for:
- `New Move`
- `Approvals Requested`
- `Approved`

## To test:
- Make sure you see only the following statuses in the TOO queue and the Status filter drop down, and that you can filter by the status.
- `New Move`
- `Service Counseling Completed`
- `Approvals Requested` 
- Make sure you can search for a move of any status, as this work shouldn't break or change the search capability.

Fix:
Before making any changes, I discovered that the New Move drop down status filter actually showed both New Move moves and Service Counseling Completed moves which was funky. I removed this _interesting_ filtering in `pkg/services/order/order_fetcher.go` and `pkg/services/move/move_searcher.go`.

Changes:
I removed the status enum values from being defined in `swagger-def/ghc.yaml`. I found this to be unnecessary as we are now handling the status filter in `pkg/handlers/ghcapi/queues.go`, which is also where we are handling the status filter for the SC and TIO roles as well.

Had to make the tests happy as well - lots of tests use "Approved" as the status for TOO actions, and that status no longer appears in the TOO queue. A few NTS tests were fixed as well. The NTS tests in question used a "Submitted" status move initially in the test generation but then moved it to "Approved" so I modified those tests to search for the move as a TOO rather than expect it in the TOO queue.
